std/algorithm/comparison.d

```diff
@@ -1565,3 +1568,39 @@ unittest
                 2, "two",
                 ));
 }
+
+/**
+    Returns true if all of the elements in $(D r1) are present in $(D r2).
+    Otherwise, returns false. 
+
+    Params:
+        r1 = Any finite input range
+        r2 = Any finite input range
+
+    Returns:
+        $(D bool)
+*/
+bool isPermutation(Range1, Range2)(Range1 r1, Range2 r2) if (
+    isInputRange!Range1 &&
+    isInputRange!Range2 &&
+    !isInfinite!Range1 && 
+    !isInfinite!Range1
+) {
+    import std.algorithm.searching : countUntil;
+
+    foreach (item; r1) {
+        if (r2.countUntil(item) == -1) {
```

Comment:
Also, this algorithm is `O(n*m)` where `n` and `m` are the lengths of the respective ranges. Surely there are more efficient algorithms that achieve the same result?